ext/strscan/strscan.c

```diff
@@ -361,6 +361,7 @@ strscan_reset(VALUE self)
 
     GET_SCANNER(self, p);
     p->curr = 0;
+    p->regex = Qnil;
```

Comment:
We need to do this in all locations that use `CLEAR_MATCH_STATUS(p)`, right?
How about adding a new macro or function that reset `p->regex` and call `CLEAR_MATCH_STATUS(p)` and replacing `CLEAR_MATCH_STATUS(p)` calls with it?

Reply:
Yes.
I will use new macro.